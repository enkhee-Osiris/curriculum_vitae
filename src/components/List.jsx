import { v4 as uuidv4 } from 'uuid';

function List({ items }) {
  return (
    <ul className="space-y-1 list-disc list-inside">
      {items.map((item) => (
        <li key={uuidv4()}>{item}</li>
      ))}
    </ul>
  );
}

export default List;
