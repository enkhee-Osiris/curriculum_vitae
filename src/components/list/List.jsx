import React from 'react';
import { arrayOf, string } from 'prop-types';

const List = ({ items }) => {
  /* eslint-disable react/no-array-index-key */
  const listItems = items.map((item, index) => (
    <li key={index}>
      {item}
    </li>));

  return <ul>{listItems}</ul>;
};

List.propTypes = {
  items: arrayOf(string).isRequired,
};

export { List as default };
