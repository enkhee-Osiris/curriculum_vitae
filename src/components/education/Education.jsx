import React, { Fragment } from 'react';
import { arrayOf, shape, string, number } from 'prop-types';
import List from '../list/List';

const Education = ({ datas }) => {
  const divs = datas.map(data => (
    <div key={data.id}>
      {data.description}
      <List items={data.items} />
    </div>
  ));

  return (
    <Fragment>
      <h3>Education</h3>
      {divs}
    </Fragment>
  );
};

Education.propTypes = {
  datas: arrayOf(shape({ id: number, description: string, items: arrayOf(string) })).isRequired,
};

export { Education as default };
