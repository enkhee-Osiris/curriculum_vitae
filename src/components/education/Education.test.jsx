import React from 'react';
import { shallow } from 'enzyme';
import Education from './Education';
import List from '../list/List';

describe('<Education />', () => {
  const props = {
    datas: [
      { id: 1, description: 'B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)', items: ['Major in Computer Science'] },
    ],
  };

  it('should be defined', () => {
    expect(Education).toBeDefined();
  });

  it('renders correctly', () => {
    const education = shallow(<Education {...props} />);

    expect(education).toMatchSnapshot();
  });

  it('renders h3', () => {
    const education = shallow(<Education {...props} />);
    const title = <h3>Education</h3>;

    expect(education.contains(title)).toEqual(true);
  });

  it('renders List', () => {
    const education = shallow(<Education {...props} />);

    expect(education.find(List).length).toBeGreaterThan(0);
  });
});
