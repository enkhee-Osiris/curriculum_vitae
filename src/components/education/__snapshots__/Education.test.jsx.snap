// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Education /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Education
    datas={
      Array [
        Object {
          "description": "B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)",
          "id": 1,
          "items": Array [
            "Major in Computer Science",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Education
        </h3>,
        Array [
          <div>
            B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)
            <List
              items={
                Array [
                  "Major in Computer Science",
                ]
              }
            />
          </div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Education",
        },
        "ref": null,
        "rendered": "Education",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)",
            <List
              items={
                Array [
                  "Major in Computer Science",
                ]
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "items": Array [
                "Major in Computer Science",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Education
          </h3>,
          Array [
            <div>
              B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)
              <List
                items={
                  Array [
                    "Major in Computer Science",
                  ]
                }
              />
            </div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Education",
          },
          "ref": null,
          "rendered": "Education",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)",
              <List
                items={
                  Array [
                    "Major in Computer Science",
                  ]
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "B.S. in Mongolian University of Science and Technology, Ulaanbaatar (2013 - 2018)",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "items": Array [
                  "Major in Computer Science",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
